File: Inter_Intent/SecondApp/app/src/main/java/com/example/secondapp/MainActivity.java
package com.example.secondapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.ComponentName;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    private EditText editText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editText = findViewById(R.id.second_edittext);

        Intent intent = getIntent();
        if(intent != null) {
            editText.setText(intent.getStringExtra("data"));
        }

        Button button = findViewById(R.id.second_button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), MainActivity.class);
                intent.putExtra("data", editText.getText().toString());
                setResult(RESULT_OK, intent);
                finish();
            }
        });
    }
}

File: app/src/main/java/com/example/mobileprogramming/MainActivity.kt
package com.example.mobileprogramming

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var count = 0
//        textview.text = count.toString()
        textview.text = "$count"

        button.setOnClickListener {
//            textview.text = (++count).toString()
            textview.text = "${++count}"
        }
    }
}